weatherman.py

```diff
@@ -0,0 +1,223 @@
+                      "\033[0;0;0m", end="")
+                print(max_temperature_value_of_a_month, end="C\n")
+
+                print("\033[1;32;34m", day_with_leading_zeros, end=" ")
+                print("+" * min_temperature_value_of_a_month,
+                      "\033[0;0;0m", end="")
+                print(min_temperature_value_of_a_month, end="C\n")
+
+                number_of_days_in_month += 1
+            break
+    del seperate_date_data
+    gc.collect()
+
+
+def bar_chart_of_temperature_everyday_of_month_in_one_line(year, month, dir_list):
+    """Searching for the year and month in files list and only opening that
+    specific file and saving weather data vlues in List and perform
+    calculations on List, after performing calculations deleting the used
+    variables, data structures and free memeory using Garbage Collector
+    """
+    min_temperature_value_of_a_month = 0
+    max_temperature_value_of_a_month = 0
+    number_of_days_in_month = 1
+    day_with_leading_zeros = ""
+    month_name = months[int(month)]
+    for file in dir_list:
+        if(year+"_"+month_name) in file:
+            f = open("./weatherfiles/"+file)
+            for line in f.readlines()[1:]:
+                seperate_date_data = line.rstrip().split(",")[:10]
+                max_temperature_value_of_a_month = integer_conversion(
+                    seperate_date_data[1])
+                min_temperature_value_of_a_month = integer_conversion(
+                    seperate_date_data[3])
+
+                # Fill the spaces of single digit with zeros so the bar chart is uniform
+                day_with_leading_zeros = str(number_of_days_in_month).zfill(2)
+                # Color the text using the ANSI Escape Sequence
+                print(day_with_leading_zeros, "\033[0;0m", end="")
+                print("\033[1;32;31m" + "+" *
+                      max_temperature_value_of_a_month + "\033[0;0m", end="")
+
+                print("\033[1;32;34m" + "+" *
+                      min_temperature_value_of_a_month + "\033[0;0m", end=" ")
+                print(max_temperature_value_of_a_month, end="C-")
+                print(min_temperature_value_of_a_month, end="C\n")
+
+                number_of_days_in_month += 1
+            break
+    del seperate_date_data
+    gc.collect()
+
+
+def main():
+    """Calling Functions according to given Command Line arguments using
+    'getopt' Library and defining the actions with each of the argument
+    after recieving
+    After calling the specific argument oriented function the returned
+    results are printed here in the specified format
+    Example: Highest Temperature: 32C on June 18, here Result with the
+    data 'June 18'
+    """
+    year = None
+    month = None
+    argv = sys.argv[1:]
+    # Exception Handling if the given command line arguments are incorrect
+    try:
+        opts, args = getopt.getopt(argv, "p:e:a:c:")
+    except:
+        print("Error")
```

Comment:
Just printing the error will not give us enough information about the error and what was the cause. We should add more information to exception handling